docs/RFCS/select_from_index.md

```diff
@@ -0,0 +1,55 @@
+- Feature Name: select_from_index
+- Status: draft
+- Start Date: 2015-08-10
+- RFC PR:
+- Cockroach Issue: #2046
+
+# Summary
+
+Extend the SELECT syntax to allow an optional index to be specified in
+table references. The current table reference syntax is `table` or
+`database.table`. This syntax would be extended to allow `table.index`
+and `database.table.index`.
+
+# Motivation
+
+As indexes are optional in SQL, it is difficult to know when an index
+is used to satisfy a query which in turn makes it difficult to use the
+existing logic test infrastructure to test correctness of index
+scans. Providing for specification of an index to read from will make
```

Comment:
Can you elaborate a bit more on what effect specifying an index will have? Does that mean my query will error out unless it can be satisfied only with lookups to the specified index(es), so that

`SELECT pk from tbl@pk WHERE name='Harold'` errors out (assuming there's only the primary key `pk`)? What about `WHERE pk > 0 AND name='Harold'`?

Also, could it make sense to have the index to use attached to the where condition? `WHERE (pk > 0 AND USEINDEX(pk)) AND name='Harold'` could mean "use the pk index for the first condition, but anything goes for the second". It just seems that that could allow for more control and support more complexity.